client/core/core.go

```diff
@@ -2632,6 +2633,84 @@ func (c *Core) Trade(pw []byte, form *TradeForm) (*Order, error) {
 	return corder, nil
 }
 
+// AccountExport is used to retrieve account by host for export.
+func (c *Core) AccountExport(pw []byte, host string) (*Account, error) {
+	_, err := c.encryptionKey(pw)
+	if err != nil {
+		return nil, codedError(passwordErr, err)
+	}
+	_, err = addrHost(host)
+	if err != nil {
+		return nil, newError(addressParseErr, "error parsing address: %v", err)
+	}
+
+	// Get the dexConnection and the dex.Asset for each asset.
+	c.connMtx.RLock()
+	dc, found := c.conns[host]
+	c.connMtx.RUnlock()
+	if !found {
+		return nil, newError(unknownDEXErr, "DEX: %s", host)
+	}
+	account := &Account{
+		Host:      host,
+		AccountID: dc.acct.id.String(),
+		PrivKey:   hex.EncodeToString(dc.acct.privKey.Serialize()),
+		PubKey:    hex.EncodeToString(dc.acct.dexPubKey.SerializeUncompressed()),
+		EncKey:    hex.EncodeToString(dc.acct.encKey),
+		Cert:      hex.EncodeToString(dc.acct.cert),
+		FeeCoin:   hex.EncodeToString(dc.acct.feeCoin),
+		IsPaid:    dc.acct.isPaid,
+	}
+	return account, nil
+}
+
+// AccountImport is used import an existing account into the db.
+func (c *Core) AccountImport(pw []byte, account Account) error {
+	_, err := c.encryptionKey(pw)
+	if err != nil {
+		return codedError(passwordErr, err)
+	}
+	_, err = addrHost(account.Host)
+	if err != nil {
+		return newError(addressParseErr, "error parsing address: %v", err)
+	}
+	var accountInfo db.AccountInfo
+	accountInfo.Host = account.Host
+	pubKey, err := hex.DecodeString(account.PubKey)
+	if err != nil {
+		return codedError(decodeErr, err)
+	}
+	accountInfo.DEXPubKey, err = secp256k1.ParsePubKey(pubKey)
+	if err != nil {
+		return codedError(parseKeyErr, err)
+	}
+	encKey, err := hex.DecodeString(account.EncKey)
+	if err != nil {
+		return codedError(decodeErr, err)
+	}
+	accountInfo.EncKey = encKey
```

Comment:
I'm not certain we want to include `EncKey`.  For a given app password and encryption key parameters stored in the DB, `EncKey` is **derived from `PrivKey`**.  Basically on import we'd be doing similar to `setupEncryption`, but with the crypter from `(*Core).encryptionKey` we'd compute an `encKey` from the imported `privKey` rather than a newly generated private key.

Reply:
Yes. The `EncKey` does encode the key parameters, but it is derived from the `PrivKey` and including both is pointless. Drop `EncKey`. If users want the backup to be encrypted, they'll have to do it themselves.

Reply:
For my own understanding, take the following in client/core/types.go, where the encKey is decrypted to get privKey:

```go
func (a *dexAccount) unlock(crypter encrypt.Crypter) error {
	keyB, err := crypter.Decrypt(a.encKey)
	...
	privKey := secp256k1.PrivKeyFromBytes(keyB)
```

and in client/core/core.go where the app password is used to create the (de)crypter when combined with keyParams:

```go
func (c *Core) encryptionKey(pw []byte) (encrypt.Crypter, error) {
	keyParams, err := c.db.Get(keyParamsKey)
	if err != nil {
		return nil, fmt.Errorf("key retrieval error: %w", err)
	}
	crypter, err := c.reCrypter(pw, keyParams)
```

The `keyParams`, which includes the salt and other encryption parameters, is combined with the app password to create the `crypter`.  The `crypter` can decrypt `encKey`.  So to decrypt an import `encKey`, you'd need to make a crypter with pw (from input) + keyParams (usually from DB).

I maybe misunderstanding, but I don't believe `EncKey` includes the key parameters, although they are generally stored in the same DB file as `EncKey` so it appears to usually be a non-issue.

Reply:
Oh right. The `EncKey` doesn't encode the parameters, the encrypted `Crypter` does.

Reply:
Now that I removed encKey I'm getting "zero-length EncKey not allowed' from: 

```go
func (db *BoltDB) CreateAccount(ai *dexdb.AccountInfo) error {
	if ai.Host == "" {
		return fmt.Errorf("empty host not allowed")
	}
	if ai.DEXPubKey == nil {
		return fmt.Errorf("nil DEXPubKey not allowed")
	}
	if len(ai.EncKey) == 0 {
		return fmt.Errorf("zero-length EncKey not allowed")
	}
```

Should I follow the steps above to put encKey in or remove the validation from CreateAccount?

Reply:
Take the crypter that's currently discarded from `_, err := c.encryptionKey(pw)` and compute `encPW` from `pw` similar to how it's done in `setupEncryption`.

Reply:
I'm noticing that both `AccountID` and `PrivKey` aren't used.  You're going to use `PrivKey` as I've described above, but the `AccountID` seems to be purely informational in the json file.  That's OK, but it's worth noting in the `AccountImport` that the `AccountID` is not validated.  To do so would involving doing similar to what `(*dexAccount).unlock` does, namely `account.AccountID == account.NewID(privKey.PubKey().SerializeCompressed())`.  Hmm, saying that, I think the `account` variable name declared in `AccountImport(pw []byte, account Account)` would be better as `acct` to avoid a collision with the `account` package.